microservices/google-api/google_search.py

```diff
@@ -382,30 +382,31 @@ def daterange(startDate, endDate):
 DROP TABLE IF EXISTS cmslite.google_pdt_old;
 
 CREATE TABLE IF NOT EXISTS cmslite.google_pdt_scratch (
-        "site"          VARCHAR(255),
-        "date"          date,
-        "query"         VARCHAR(2048),
-        "country"       VARCHAR(255),
-        "device"        VARCHAR(255),
-        "page"          VARCHAR(2047),
-        "position"      FLOAT,
-        "clicks"        DECIMAL,
-        "ctr"           FLOAT,
-        "impressions"   DECIMAL,
-        "node_id"       VARCHAR(255),
-        "page_urlhost"  VARCHAR(255),
-        "title"         VARCHAR(2047),
-        "theme_id"      VARCHAR(255),
-        "subtheme_id"   VARCHAR(255),
-        "topic_id"      VARCHAR(255),
-        "subtopic_id"      VARCHAR(255),
-        "subsubtopic_id"      VARCHAR(255),
-        "theme"         VARCHAR(2047),
-        "subtheme"      VARCHAR(2047),
-        "topic"         VARCHAR(2047),
-        "subtopic"         VARCHAR(2047),
-        "subsubtopic"         VARCHAR(2047)
-);
+        site              VARCHAR(255)    ENCODE ZSTD,
+        date              date            ENCODE AZ64,
+        query             VARCHAR(2048)   ENCODE ZSTD,
+        country           VARCHAR(255)    ENCODE ZSTD,
+        device            VARCHAR(255)    ENCODE ZSTD,
+        page              VARCHAR(2047)   ENCODE ZSTD,
+        position          FLOAT           ENCODE ZSTD,
+        clicks            DECIMAL         ENCODE ZSTD,
+        ctr               FLOAT           ENCODE ZSTD,
+        impressions       DECIMAL         ENCODE ZSTD,
+        node_id           VARCHAR(255)    ENCODE ZSTD,
+        page_urlhost      VARCHAR(255)    ENCODE ZSTD,
+        title             VARCHAR(2047)   ENCODE ZSTD,
+        theme_id          VARCHAR(255)    ENCODE ZSTD,
+        subtheme_id       VARCHAR(255)    ENCODE ZSTD,
+        topic_id          VARCHAR(255)    ENCODE ZSTD,
+        subtopic_id       VARCHAR(255)    ENCODE ZSTD,
+        subsubtopic_id    VARCHAR(255)    ENCODE ZSTD,
+        theme             VARCHAR(2047)   ENCODE ZSTD,
+        subtheme          VARCHAR(2047)   ENCODE ZSTD,
+        topic             VARCHAR(2047)   ENCODE ZSTD,
+        subtopic          VARCHAR(2047)   ENCODE ZSTD,
+        subsubtopic       VARCHAR(2047)   ENCODE ZSTD)
+        COMPOUND SORTKEY (date,page_urlhost,theme,node_id);
```

Comment:
Since not all sites have `node_id` but all have a `page`, I'm replacing `node_id` with `page`.

In embed dashboards, we still use `theme` enough that I think it's valuable to sort on; and any site that don't have a `theme` will just result in a zone of NULL `theme`.

Then sorting on `page` will have sorted pages under a sort of either their theme name, or the theme=NULL.

Finally, I'm going to add clicks, since in all our explores, we always sort on `COALESCE(SUM(google_search.clicks), 0) AS "google_search.total_clicks"` descending, and so sorting on clicks will reduce sort steps in most queries on this table.

Sorting will almost never result in worse performance; it will in the worst case result in slower table insertions, which we only do once daily anyway. So, we can add more compound sort keys as we need